packages/shylu/shylu_dd/frosch/src/InterfaceSets/FROSch_DDInterface_def.hpp

```diff
@@ -547,59 +547,196 @@ namespace FROSch {
+                                                             GOVecVec &componentsSubdomainsUnique,
+                                                             UN priorDofsPerNode)
     {
-        Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMat = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(NodesMap_,10);
-        Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMatTmp = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(UniqueNodesMap_,10);
+        
+
+        if (UniqueNodesMap_->getMinAllGlobalIndex() > 0 || priorDofsPerNode > 0) {
+            GO minGID = UniqueNodesMap_->getMinAllGlobalIndex();
+            ConstGOVecView ElementList;
+            MapPtr tmpUniqueNodesMap;
+            {
+                GOVec indices(UniqueNodesMap_->getNodeNumElements());
+                ElementList = UniqueNodesMap_->getNodeElementList();
+                for (UN i=0; i<indices.size(); i++) {
+                    indices[i] = ElementList[i] - minGID;
+                }
+                tmpUniqueNodesMap = Xpetra::MapFactory<LO,GO,NO>::Build(UniqueNodesMap_->lib(),-1,indices(),0,UniqueNodesMap_->getComm());
+            }
+
+            MapPtr tmpNodesMap;
+            {
+                GOVec indices(NodesMap_->getNodeNumElements());
+                ElementList = NodesMap_->getNodeElementList();
+                for (UN i=0; i<indices.size(); i++) {
+                    indices[i] = ElementList[i] - minGID - (GO) priorDofsPerNode;
+                }
+                tmpNodesMap = Xpetra::MapFactory<LO,GO,NO>::Build(NodesMap_->lib(),-1,indices(),0,NodesMap_->getComm());
+            }
+
+            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMat = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpNodesMap,10);
+            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMatTmp = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpUniqueNodesMap,10);
 #ifdef Tpetra_issue_1752
-        // AH 10/10/2017: Can we get away with using just one importer/exporter after the Tpetra issue is fixed?
-        Teuchos::RCP<Xpetra::Import<LO,GO,NO> > commImporter = Xpetra::ImportFactory<LO,GO,NO>::Build(UniqueNodesMap_,NodesMap_);
-        Teuchos::RCP<Xpetra::Export<LO,GO,NO> > commExporter = Xpetra::ExportFactory<LO,GO,NO>::Build(NodesMap_,UniqueNodesMap_);
+            // AH 10/10/2017: Can we get away with using just one importer/exporter after the Tpetra issue is fixed?
+            Teuchos::RCP<Xpetra::Import<LO,GO,NO> > commImporter = Xpetra::ImportFactory<LO,GO,NO>::Build(tmpUniqueNodesMap,tmpNodesMap);
+            Teuchos::RCP<Xpetra::Export<LO,GO,NO> > commExporter = Xpetra::ExportFactory<LO,GO,NO>::Build(tmpNodesMap,tmpUniqueNodesMap);
 #else
-        Teuchos::RCP<Xpetra::Export<LO,GO,NO> > commExporter = Xpetra::ExportFactory<LO,GO,NO>::Build(NodesMap_,UniqueNodesMap_);
+            Teuchos::RCP<Xpetra::Export<LO,GO,NO> > commExporter = Xpetra::ExportFactory<LO,GO,NO>::Build(tmpNodesMap,tmpUniqueNodesMap);
 #endif
```

Comment:
Do the `if` and the `else` block both work? If so, I think we should finally remove the `if` block.